app/views/repositories/_form.html.erb

```diff
@@ -94,3 +108,11 @@
   <%= f.submit t('save'), class: 'btn btn-primary' %>
   <%= link_to t('back'),  project_path(@project_id), class: 'btn btn-default' %>
 </div>
+
+<script type="text/javascript">
+  var _repository_branch = new Repository.Branch;
```

Comment:
Actually somehow JS vars can't hava the same name as HTML element IDs. So I had to add the underscore in order to preserve the good var name and avoid conflicts.